quotes_recommender/vector_store/vector_store_qdrant.py

```diff
@@ -175,10 +176,46 @@ def get_similarity_scores(self, query_embedding: npt.NDArray[np.float64]) -> Opt
         Reference: https://qdrant.github.io/qdrant/redoc/index.html#tag/points/operation/search_points
         """
         dups = self.client.search(
-            collection_name='quotes',
+            collection_name=DEFAULT_QUOTE_COLLECTION,
             query_vector=query_embedding,
             limit=1,
             score_threshold=0.9,
         )
         # return payload results
         return dups
+
+    def get_entry_by_author(
+        self,
+        query_embedding: npt.NDArray[np.float64],
+        author: str,
+        collection: str = DEFAULT_QUOTE_COLLECTION,
+    ) -> Optional[list[ScoredPoint]]:
+        """
+        Get entry with the same author based on similarity scores for the specified query.
+        :param author: Author to match.
+        :param query_embedding: Encoded quote to be checked for duplicate detection.
+        :param collection: Collection used for the search.
+        :return: Payload results of quotes with the same author.
+
+        Reference: https://qdrant.github.io/qdrant/redoc/index.html#tag/points/operation/search_points
+        """
+        # build the search query
+        result = self.client.search(
+            collection_name=collection,
+            query_vector=query_embedding,
+            # Add a condition to match the author
+            query_filter=models.Filter(
+                must=[
+                    models.FieldCondition(
+                        key='author',
+                        match=models.MatchValue(
+                            value=author,
+                        ),
+                    )
+                ]
+            ),
+            limit=1,
+            score_threshold=0,
+        )
+        # return payload results
+        return result
```

Comment:
Ich habe das mal mit result[0] ausprobiert, aber ich erhalte einen Index out of range error: weil das aktuelle result objekt ein vordefiniertes ScoredPoint Objekt ist... 
vgl.
`[ScoredPoint(id=1211685, version=809, score=1.0, payload={'author': 'Brennan Manning', 'likes': 166, 'quote': 'I have a ...should be. Because you’re never going to be as you should be.', 'tags': ['prayer', 'love you', 'past']}, vector=None)]`